libs/pubsub/subindex.go

```diff
@@ -0,0 +1,102 @@
+package pubsub
+
+import "github.com/tendermint/tendermint/abci/types"
+
+// An item to be published to subscribers.
+type item struct {
+	Data   interface{}
+	Events []types.Event
+}
+
+// A subInfo value records a single subscription.
+type subInfo struct {
+	clientID string        // chosen by the client
+	query    Query         // chosen by the client
+	subID    string        // assigned at registration
+	sub      *Subscription // receives published events
```

Comment:
if the set type includes this doesn't this mean we can have duplicate subscriptions for the same data stream? do we want to support that.